scripts/locust/service.py

```diff
@@ -40,120 +81,251 @@ def parallel_requests():
+    """
+
+    @task
+    def _choose_file(self):
+        super().choose_file({"fileFormat": {"is": ["fastq", "fastq.gz"]}})
+
+    def _download(self, path: str):
+        url = self.endpoint(path.format(file_uuid=self.file_uuid))
+        self.download_redirects(url, path)
+
+    @task
+    def file(self):
+        self._download('/repository/files/{file_uuid}')
+
+    @task
+    def fetch_file(self):
+        self._download('/fetch/repository/files/{file_uuid}')
+
+
+class GA4GHTaskSet(FileSelectionTaskSet):
+    """
+    Test DRS and DOS endpoints after selecting a file.
+    """
+
+    # Our DRS implementation only works for files in the DCP1 catalog
+    catalog = 'dcp1'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.drs_access_ids = set()
 
     @task
-    class SamplesTaskSet(TaskSet):
-
-        def on_start(self):
-            self.samples_page()
-
-        def samples_page(self):
-            with parallel_requests() as group:
-                group.spawn(lambda: self.client.get('/index/summary?filters=%7B%7D'))
-                group.spawn(lambda: self.client.get('/index/samples?filters=%7B%7D&size=15'))
-
-        @seq_task(1)
-        def select_brain(self):
-            with parallel_requests() as group:
-                group.spawn(lambda: self.client.get('/index/summary?filters=%7B%22file%22%3A%7B%22organ%22'
-                                                    '%3A%7B%22is%22%3A%5B%22brain%22%5D%7D%7D%7D'))
-                group.spawn(lambda: self.client.get('/index/samples?filters=%7B%22file%22%3A%7B%22organ%22'
-                                                    '%3A%7B%22is%22%3A%5B%22brain%22%5D%7D%7D%7D'
-                                                    '&size=15&sort=sampleId&order=desc'))
-
-        @seq_task(2)
-        def next_page_1(self):
-            self.client.get('/index/samples?filters=%7B%22file%22%3A%7B%22organ%22'
-                            '%3A%7B%22is%22%3A%5B%22brain%22%5D%7D%7D%7D&size=15'
-                            '&sort=sampleId&order=desc&search_after=Q4_DEMO-'
-                            'sample_SAMN02797092&search_after_uid=doc'
-                            '%23e8dcd716-03d2-4244-a196-b7269b5e5e6f')
-
-        @seq_task(3)
-        def next_page_2(self):
-            self.client.get('/index/samples?filters=%7B%22file%22%3A%7B%22organ%22%'
-                            '3A%7B%22is%22%3A%5B%22brain%22%5D%7D%7D%7D&size=15'
-                            '&sort=sampleId&order=desc&search_after=Q4_DEMO-'
-                            'sample_SAMN02797092&search_after_uid=doc'
-                            '%23da9bd051-9ce7-4a38-99c1-284112f0f483')
-
-        @seq_task(4)
-        def stop(self):
-            self.interrupt()
-
-
-class ServiceLocust(HttpLocust):
-    host = 'https://service.dev.singlecell.gi.ucsc.edu'
-    task_set = ServiceTaskSet
-    min_wait = 1000
-    max_wait = 5000
+    def _choose_file(self):
```

Comment:
Use of super() is odd since a different method is called, not the overridden one.